src/host.ps1

```diff
@@ -75,88 +108,89 @@ function Get-XoHost {
-                    # Apply limit if specified
-                    $processLimit = if ($Limit -gt 0) { [Math]::Min($Limit, $hostUrls.Count) } else { $hostUrls.Count }
-                    $processUrls = $hostUrls | Select-Object -First $processLimit
-                    
-                    foreach ($hostUrl in $processUrls) {
-                        # Skip null or empty URLs
-                        if ([string]::IsNullOrEmpty($hostUrl)) {
-                            continue
-                        }
-                        
-                        try {
-                            # Extract the ID from the URL string
-                            $match = [regex]::Match($hostUrl, "\/rest\/v0\/hosts\/([^\/]+)$")
-                            if ($match.Success) {
-                                $id = $match.Groups[1].Value
-                                if (![string]::IsNullOrEmpty($id)) {
-                                    $hostDetail = Invoke-RestMethod -Uri "$script:XoHost$hostUrl" @script:XoRestParameters
+    try {
+        $uri = "$script:XoHost/rest/v0/hosts"
+        Write-Verbose "Getting list of all hosts from $uri"
+        
+        $allHostsResponse = Invoke-RestMethod -Uri $uri @script:XoRestParameters
+        Write-Verbose "API response received with $($allHostsResponse.Count) total hosts"
+
+        $hostsToProcess = $allHostsResponse
+        if ($Limit -gt 0 -and $allHostsResponse.Count -gt $Limit) {
+            $hostsToProcess = $allHostsResponse[0..($Limit-1)]
+            Write-Verbose "Processing first $Limit hosts out of $($allHostsResponse.Count) total"
+        } else {
+            Write-Verbose "Processing all $($allHostsResponse.Count) hosts"
+        }
+        
+        $processedCount = 0
+        
+        foreach ($hostUrl in $hostsToProcess) {
+            $cleanUrl = $hostUrl.Trim('",\[\] \n\r\t')
+            if ([string]::IsNullOrEmpty($cleanUrl)) {
+                continue
+            }
 
-                                    if ($hostDetail) {
-                                        # Parse the response if it's a string
-                                        if ($hostDetail -is [string]) {
-                                            $hostDetail = $hostDetail | ConvertFrom-Json -AsHashtable
-                                        }
-                                        
-                                        ConvertTo-XoHostObject $hostDetail
-                                    }
-                                }
-                                else {
-                                    Write-Warning "Failed to extract valid ID from URL: $hostUrl"
-                                }
-                            }
-                        } catch {
-                            Write-Warning "Failed to retrieve host details from URL: $hostUrl. Error: $_"
+            if ($cleanUrl -match "/hosts/([0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})") {
```

Comment:
Why the trim and match? Also, you can already request all host-relevant fields with one API call, why make one request per host?